### PR TITLE
Add defaults to restore backwards compatibility

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

When we removed config.default.json we broke the upgrade path for
projects without explicit settings for sms. This restores the defaults
in code.

medic/cht-core#6163

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
Review please @dianabarsan 

I went through one of the last copies of the default config and I think these are the only parts where the code didn't correctly handle the undefined settings.

The three options I thought of were:
1. Adding them to the new default config. This would work for new installs but not upgrades and it's not merged with existing configs.
2. Adding a migration to merge the default values on upgrade and fresh install. This would have worked ok but it's a bit heavy handed for what it is.
3. Hardcoding defaults into the app (the one I picked).

Thanks for your time!

File: api/src/services/messaging.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ const applyTaskStateChangesToDocs = (taskStateChanges, docs) => {
 };
 
 const getOutgoingMessageService = () => {
-  const settings = config.get('sms');
+  const settings = config.get('sms') || { sms: { outgoing_service: 'medic-gateway' } };
```
Comment (dianabarsan) [inline api/src/services/messaging.js]: I think we also need to add the default value to the `isMedicGatewayEnabled` function from the same service. 
The method is called by `sms-gateway` controller when passing outgoing messages to `/api/sms`. Without it, it always returns an empty list.

Comment (garethbowen) [reply api/src/services/messaging.js]: My search and replace skills failed me... Fixed!

Comment (garethbowen) [reply api/src/services/messaging.js]: Also, the default object is wrong - it should just be `{ outgoing_service: 'medic-gateway' }`. SMH.

**Pull request merged**